Userland/Libraries/LibJS/Runtime/Temporal/DurationPrototype.cpp

```diff
@@ -286,6 +288,110 @@ JS_DEFINE_NATIVE_FUNCTION(DurationPrototype::abs)
+    }
+
+    // 10. Let balanceResult be ? BalanceDuration(unbalanceResult.[[Days]], duration.[[Hours]], duration.[[Minutes]], duration.[[Seconds]], duration.[[Milliseconds]], duration.[[Microseconds]], duration.[[Nanoseconds]], unit, intermediate).
+    auto balance_result = TRY(balance_duration(global_object, unbalance_result.days, duration->hours(), duration->minutes(), duration->seconds(), duration->milliseconds(), duration->microseconds(), *js_bigint(vm, Crypto::SignedBigInteger::create_from(duration->nanoseconds())), unit, intermediate));
+
+    // 11. Let roundResult be ? RoundDuration(unbalanceResult.[[Years]], unbalanceResult.[[Months]], unbalanceResult.[[Weeks]], balanceResult.[[Days]], balanceResult.[[Hours]], balanceResult.[[Minutes]], balanceResult.[[Seconds]], balanceResult.[[Milliseconds]], balanceResult.[[Microseconds]], balanceResult.[[Nanoseconds]], 1, unit, "trunc", relativeTo).
+    auto round_result = TRY(round_duration(global_object, unbalance_result.years, unbalance_result.months, unbalance_result.weeks, balance_result.days, balance_result.hours, balance_result.minutes, balance_result.seconds, balance_result.milliseconds, balance_result.microseconds, balance_result.nanoseconds, 1, unit, "trunc"sv, relative_to.is_object() ? &relative_to.as_object() : nullptr));
+
+    double whole;
+
+    // 12. If unit is "year", then
+    if (unit == "year"sv) {
+        // a. Let whole be roundResult.[[Years]].
+        whole = round_result.years;
+    }
+
+    // 13. If unit is "month", then
+    if (unit == "month"sv) {
+        // a. Let whole be roundResult.[[Months]].
+        whole = round_result.months;
+    }
+
+    // 14. If unit is "week", then
+    if (unit == "week"sv) {
+        // a. Let whole be roundResult.[[Weeks]].
+        whole = round_result.weeks;
+    }
+
+    // 15. If unit is "day", then
+    if (unit == "day"sv) {
+        // a. Let whole be roundResult.[[Days]].
+        whole = round_result.days;
+    }
+
+    // 16. If unit is "hour", then
+    if (unit == "hour"sv) {
+        // a. Let whole be roundResult.[[Hours]].
+        whole = round_result.hours;
+    }
+
+    // 17. If unit is "minute", then
+    if (unit == "minute"sv) {
+        // a. Let whole be roundResult.[[Minutes]].
+        whole = round_result.minutes;
+    }
+
+    // 18. If unit is "second", then
+    if (unit == "second"sv) {
+        // a. Let whole be roundResult.[[Seconds]].
+        whole = round_result.seconds;
+    }
+
+    // 19. If unit is "millisecond", then
+    if (unit == "millisecond"sv) {
+        // a. Let whole be roundResult.[[Milliseconds]].
+        whole = round_result.milliseconds;
+    }
+
+    // 20. If unit is "microsecond", then
+    if (unit == "microsecond"sv) {
+        // a. Let whole be roundResult.[[Microseconds]].
+        whole = round_result.microseconds;
+    }
+
+    // 21. If unit is "nanosecond", then
+    if (unit == "nanosecond"sv) {
```

Comment:
It's a bit annoying that the we have to do all of these ifs unconditionally even if only one of them can be true, but I guess that's a spec issue

Reply:
yeah, these would be nicer as `else-if`s. Should be a simple editorial change, I'll make a PR :)

Reply:
https://github.com/tc39/proposal-temporal/pull/1918